src/storage/txn/process.rs

```diff
@@ -147,7 +157,13 @@ impl<E: Engine> Executor<E> {
     }
 
     /// Start the execution of the task.
-    pub fn execute(mut self, cb_ctx: CbContext, snapshot: EngineResult<E::Snap>, task: Task) {
+    pub fn execute(
+        mut self,
+        cb_ctx: CbContext,
+        snapshot: EngineResult<E::Snap>,
+        task: Task,
+        storage_cb: Arc<Mutex<Option<StorageCb>>>,
```

Comment:
em, seem we can use crossbeam AtomicOption

Reply:
Good, i will have a try.

Reply:
crossbeam 0.5 abandoned AtomicOption.

Reply:
oh, bad news, AtomicOptino looks better...
@brson do you have any other suggestion?

Reply:
You could use `AtomicCell<Option<T>>` in place of `AtomicOption<T>` (see crossbeam-rs/crossbeam#186)

Reply:
AtomicCell looks like the right choice. I'd also assert that AtomicCell::is_lock_free, to make sure that `Option<StorageCb>` stays under a word long.